docs/testing_servers.md

```diff
@@ -0,0 +1,340 @@
+# Testing Servers
+
+Testing servers with the conformance runner involves the following steps:
+
+* Implementing the ConformanceService endpoints.
+* Making an executable file that will read messages from `stdin` and write messages to `stdout`.
+* Defining any configuration for what your server supports. For more information on how to do this, see the [Configuration Files](./configuring_and_running_tests.md#configuration-files) section in the [Configuring and Running Tests documentation](./configuring_and_running_tests.md).
+
+## Process
+
+The basic process for working with servers in the conformance suite will work as follows:
+
+* The conformance runner will pass a `ServerCompatRequest` message via `stdin`. This message will contain all the details necessary
+  for your implementation to start its server-under-test. Note that the request does not specify a port to listen on.
+  Implementations should instead pick an available ephemeral port according to their OS and return that value in the response.
+
+  Fields in the request are:
+
+  * `protocol` specifies what protocol will be used to run the tests. 
+  * `http_version` specifies which HTTP version will be used.
+  * `use_tls` specifies whether your server should generate a self-signed certificate. Clients will be configured to trust
```

Comment:
FWIW, it is not _required_ for the cert to be self-signed. It can be any TLS certificate where the subject matches the host name that gets sent back in the `ServerCompactResponse`.

But we explicitly _allow_ self-signed (and "localhost" as the subject) to make it easier to implement the server w/out having to worry about certificate issuance, managing/distributing certificate secrets, etc.